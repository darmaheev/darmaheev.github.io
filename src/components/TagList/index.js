import React from "react";
import PropTypes from "prop-types";

import TagItem from "../TagItem";
import {
  Wrapper
} from "./styles";

const TagList = ({
  tags,
  handleDelete
}) => {
  return (
    <Wrapper center={typeof handleDelete !== "function"}>
      {tags.map(tag => (<TagItem key={tag} handleDelete={handleDelete}>{tag}</TagItem>))}
    </Wrapper>
  );
};

TagList.defaultProps = {
  tags: [],
  handleDelete: undefined
};

TagList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string),
  handleDelete: PropTypes.func
};

export default TagList;