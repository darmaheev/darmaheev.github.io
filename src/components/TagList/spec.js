import React from "react";
import { MemoryRouter } from "react-router-dom";
import TagList from "./index";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders tagList with default props", () => {
  const tagList = shallow(
    <TagList/>
  );
  expect(tagList).toMatchSnapshot();
});

it("renders tagList with data", () => {
  const tags = ["tag1", "tag2"];

  const tagList = mount(
    <MemoryRouter>
      <TagList
        tags={tags}
      />
    </MemoryRouter>
  );
  expect(tagList.text()).toContain("tag1");
  expect(tagList.text()).toContain("tag2");
});

it("check tagList on click", () => {
  const tags = ["tag1", "tag2"];
  const handleDelete = jest.fn();

  const tagList = mount(
    <MemoryRouter>
      <TagList
        tags={tags}
        handleDelete={handleDelete}
      />
    </MemoryRouter>
  );
  const deleteControl = tagList.find("button").at(1);
  deleteControl.simulate("click");
  expect(handleDelete).toHaveBeenCalledWith("tag2");
});

