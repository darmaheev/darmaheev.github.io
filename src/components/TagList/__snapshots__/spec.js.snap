// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tagList with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagList
    tags={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "center": true,
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  ",
          [Function],
          "

",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "center": true,
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            "
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  ",
            [Function],
            "

",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
